mitxgraders/helpers/validatorfuncs.py

```diff
@@ -83,6 +84,49 @@ def is_callable(obj):
     """Returns true if obj is callable"""
     return callable(obj)
 
+def is_callable_with_args(num_args):
+    """
+    Validates that a function is callable and takes num_args arguments
+
+    Examples:
+    >>> def func(x, y): return x + y
+    >>> is_callable_with_args(2)(func) == func
+    True
+    >>> is_callable_with_args(3)(func) == func # doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    Invalid: Expected function... to have 3 arguments, instead it has 2
+
+    Callable objects work, too:
+    >>> class Foo:
+    ...     def __call__(self, x):
+    ...         return x
+    >>> foo = Foo()
+    >>> is_callable_with_args(1)(foo) == foo
+    True
+    >>> is_callable_with_args(1)(Foo) # doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    Invalid: Expected function... to have 1 arguments, instead it has 2
+    """
+    def _validate(func):
+        # first, check that the function is callable
+        is_callable(func)
+        try:
+            # assume object is a function
+            f_args = len(getargspec(func)[0])
+        except TypeError:
+            # otherwise it is a callable object
+            f_args = len(getargspec(func.__call__)[0])
+            # If obj.__call__ is a bound method, remove one argument
+            if func.__call__.__self__ is not None:
+                f_args += -1
+        if not f_args == num_args:
+            msg = "Expected function {func} to have {num_args} arguments, instead it has {f_args}"
+            raise Invalid(msg.format(func=func, num_args=num_args, f_args=f_args))
+        return func
+
+    return _validate
+
```

Comment:
I hope this function is fully tested!

Is there a reason to make this function return another function, rather than saying `is_callable_with_nargs(func, 3)`?

Reply:
I think that voluptuous requires its validation functions to take a single argument, which is why `is_callable_with_args` returns a function. 

The tests for `is_callable_with_args` are doctests in the docstring. I'm going to comment them a little bit better.

Reply:
Ok. Can you add a test to make sure that 120/121 are triggered too? That's the one that I know least about.